lib/wanda/execution/evaluation.ex

```diff
@@ -18,44 +18,52 @@ defmodule Wanda.Execution.Evaluation do
   }
 
   # Abacus spec is wrong
-  @dialyzer {:nowarn_function, eval_expectation: 2}
+  @dialyzer {:nowarn_function, eval_expectation: 3, value_from_conditions: 3}
 
-  @spec execute(String.t(), String.t(), [Check.t()], map()) :: Result.t()
-  def execute(execution_id, group_id, checks, gathered_facts, timeouts \\ []) do
+  @spec execute(
+          String.t(),
+          String.t(),
+          [Check.t()],
+          %{String.t() => boolean() | number() | String.t()},
```

Comment:
The signature looks wrong. You have changed the order between `gathered_facts` (which was a map) with the new `env`.

Reply:
Right. I believe I got confused by the missing type for `timeouts`.